src/SFA.DAS.FAT.Web.UnitTests/Services/SessionServiceTests.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System.Text;
+using System.Text.Json;
+using AutoFixture.NUnit3;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.FAT.Web.Services;
+
+namespace SFA.DAS.FAT.Web.UnitTests.Services;
+
+public class SessionServiceTests
+{
+    private Mock<ISession> _sessionMock;
+    private SessionService _sut;
+
+    [SetUp]
+    public void Before_Each_Test()
+    {
+        _sessionMock = new Mock<ISession>();
+        DefaultHttpContext httpContext = new()
+        {
+            Session = _sessionMock.Object
+        };
+        var httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+        httpContextAccessorMock.Setup(h => h.HttpContext).Returns(httpContext);
+        _sut = new SessionService(httpContextAccessorMock.Object);
+    }
+
+    [Test, AutoData]
+    public void Set_AddsValueToSession(string key, string value)
+    {
+        _sut.Set(key, value);
+
+        _sessionMock.Verify(s => s.Set(key, Encoding.UTF8.GetBytes(value)));
+    }
+
+    [Test, AutoData]
+    public void Set_AddsValueToSessionAndOverridesAnyExistingValue(string key, string value, string newValue)
```

Comment:
This test doesn't actually show that the old value is overwritten, just the both values have been written. Happy if that's the desired test functionality but in that case the test name is wrong